src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -4330,28 +4321,43 @@ namespace HeatBalanceManager {
                                "of above errors",
                                DesiredConstructionName));
 
+                for (int iPhi = 0; iPhi < Window::numPhis; ++iPhi) {
+                    Tsol[iPhi] = TsolTemp(iPhi+1);
+                    Tvis[iPhi] = TvisTemp(iPhi+1);
+                    Rfsol[iPhi] = RfsolTemp(iPhi+1);
+                    Rbsol[iPhi] = RbsolTemp(iPhi+1);
+                    Rfvis[iPhi] = RfvisTemp(iPhi+1);
+                    Rbvis[iPhi] = RbvisTemp(iPhi+1);
+                }
+
+                for (IGlass = 1; IGlass <= NGlass(IGlSys); ++IGlass) {
+                    for (int iPhi = 0; iPhi < Window::numPhis; ++iPhi) {
+                         AbsSol(IGlass)[iPhi] = AbsSolTemp(IGlass, iPhi+1);
+                    }
+                }
+                        
                 // Hemis
-                thisConstruct.TransDiff = Tsol(11);
-                thisConstruct.TransDiffVis = Tvis(11);
-                thisConstruct.ReflectSolDiffFront = Rfsol(11);
-                thisConstruct.ReflectSolDiffBack = Rbsol(11);
-                thisConstruct.ReflectVisDiffFront = Rfvis(11);
-                thisConstruct.ReflectVisDiffBack = Rbvis(11);
-
-                Window::W5LsqFit(CosPhiIndepVar, Tsol, 6, 1, 10, thisConstruct.TransSolBeamCoef);
-                Window::W5LsqFit(CosPhiIndepVar, Tvis, 6, 1, 10, thisConstruct.TransVisBeamCoef);
-                Window::W5LsqFit(CosPhiIndepVar, Rfsol, 6, 1, 10, thisConstruct.ReflSolBeamFrontCoef);
+                thisConstruct.TransDiff = TsolTemp(11);
```

Comment:
There is also this issue where a line in the W5 file contains both an array and a scalar variable and a single array is used to read both. Pffft.